File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        s_hash_map = {}
        t_hash_map = {}

        for char in s:
            s_hash_map[char] = s_hash_map.get(char, 0) + 1

        for char in t:
            t_hash_map[char] = t_hash_map.get(char, 0) + 1

        return s_hash_map == t_hash_map      


                

          

File: 0242-valid-anagram/0242-valid-anagram.dart
class Solution {
  bool isAnagram(String s, String t) {
       Map<String, int> freqMap = {};

       if(s.length != t.length) return false;

       for(String c in s.split('')) {
          freqMap[c] = (freqMap[c] ?? 0) + 1;
       }

       for(String c in t.split('')) {
          freqMap[c] = (freqMap[c] ?? 0) - 1;
       }

    return freqMap.values.every((v) => v == 0);
  }
}